### ISSUE TITLE
No sound on Dell Latitude D530

### ISSUE THREAD

**Issue by bert003** [Baron-von-Riedesel/VSBHDA#3]
Hi,

Everything loads fine on my Dell Latitude. However I get no sound in Doom or Keen Dreams (the only 2 games I tried). Refer to the image below. Any suggestions please?

![image](https://github.com/Baron-von-Riedesel/VSBHDA/assets/20399702/1e2c149f-b70f-4053-9076-5c77b0fdea4b)

**Comment by bert003** [comment]
Using PCITools give me this:

0 1b  0       4  3  0  2  8086/284b   **9**  multimedia controller (HDA)

However I cannot set IRQ 9 in right?

**Comment by Baron-von-Riedesel** [comment]
IRQ 9 for the HDA controller is ok - the only thing to care for is that the emulated SB IRQ ( 5 or 7 ) must not match the HDA IRQ.

If your machine is a notebook, the /O option should be used to change output to speaker or headphone.

**Comment by bert003** [comment]
I tried all option with /O. I also changed the address, irq and dma. No luck.

If I use the headphones, I hear a very short tick (whichever /O option I use) but then pressing CTRL-G to test the beep gives nothing.

Also, Keen Dreams does not detect the SB but even setting it to ADLIB gives no sound. See image below.

![image](https://github.com/Baron-von-Riedesel/VSBHDA/assets/20399702/999711ba-c972-4f1c-84fb-6d3a6eb37c5b)

Without loading VSBHDA the beep is heard from the PC Speaker and even games use the PC Speaker.

The original SBEMU does not even load but just hangs on boot. So I would really like to make this work.

Is there anything else I can test please?

**Comment by Baron-von-Riedesel** [comment]
> Is there anything else I can test please?

Yes. Try Mpxplay or HDAPlay ( contained in [HDAUtils](url) ) - just to ensure that the HDA controller works.

**Comment by tedkotz** [comment]
The default volume level can be pretty low. 
Try the /VOL9 option.
If you are looking for something to try/test with:
To test realmode the shareware version of wolf3d works well.
To test protected mode the share version of doom is pretty sharp
The open source Boom shipped with FreeDOS includes allegro asetup.exe which is great for proof.

**Comment by bert003** [comment]
Thanks. Will test later on this week and revert.

In the meantime, I have tested on a PC (not laptop) and everything worked as it should.

So I guess that the sound is there on the laptop but, for some strange reason, cannot be heard.

**Comment by bert003** [comment]
Attaching the output from HDASTAT.EXE.... can you help me interpret it correctly please?

[HDASTAT.TXT](https://github.com/Baron-von-Riedesel/VSBHDA/files/13372729/HDASTAT.TXT)

**Comment by Baron-von-Riedesel** [comment]
> can you help me interpret it correctly please?

The things that are most interesting are behind "pin complex". Checking the fields "def. device", there's indeed no lineout pin, only HP ( headphone ) and speaker.

Did you try to play the .WAV file with HDAPLAY? If yes, then don't load SBEMU/VSBHDA - HDAPLAY communicates with the HDA directly, no SB emulation is required.

**Comment by bert003** [comment]
Yes as in my previous post (OUTPUT.TXT) I played the WAV file with VSBHDA loaded.

Now I played it without VSBHDA loaded and got the following:

![image](https://github.com/Baron-von-Riedesel/VSBHDA/assets/20399702/ca6dc743-bc7a-4c6f-996d-d0ac49910e8b)

**Comment by Baron-von-Riedesel** [comment]
> Now I played it without VSBHDA loaded and got the following:

HDAPLAY tries to select "lineout" as default. With option -s, it selects "speaker". "Headphone" cannot be selected currently.

**Comment by bert003** [comment]
I will try the -s option and let you know.

In the meantime, should the -s option not be successful, what is the next step please?

**Comment by bert003** [comment]
There is no -s in hdaplay. Any specific version I should download?

I only have -r and -q

**Comment by Baron-von-Riedesel** [comment]
> There is no -s in hdaplay. Any specific version I should download?

The current source has the -s option already, but the binary ( contained in the "tag"-zip ) is created from older version.

So you'll have to create the current binary version on your own. Shouldn't be too hard ...

**Comment by bert003** [comment]
Thanks so much for taking the time to provide the binaries.

Downloaded **The Microsoft Sound.wav** from [here](https://winsounds.com/windows-95-sounds/) and tested with the -s.

Kindly note below. This time it seemed to have played but all I hear is a short tick sound similar to the one I hear on boot.

![image](https://github.com/Baron-von-Riedesel/VSBHDA/assets/20399702/5903b560-5dc5-467f-90ce-b19ec19961dc)

Any suggestions please?

**Comment by Baron-von-Riedesel** [comment]
> Any suggestions please?

Yes. Try a .wav file with CD attributes ( 2 channels, 16 bits/sample, 44.1 kHz ) - that's the format that I tested when this little test tool was written.

**Comment by bert003** [comment]
May I kindly ask you to provide a sample WAV file please? I downloaded two files and one showed "No data chunk found" and the other showed  "format not supported". Thanks in advance.

**Comment by Baron-von-Riedesel** [comment]
HDAPLAY wants to access the HDA controller exclusively - hence cannot run concurrently with SBEMU/VSBHDA.

I checked your sample1.zip - it's not really a "beep", sounds more like a simple sinusoidal wave.

btw, lots of tools ( i.e. audacity ) can be used to convert any MP3 to WAV, stereo, 16bit, 44.1 kHz.

There are no more tests that I can supply.

**Comment by bert003** [comment]
Just wanted to add that the sound card is SigmaTel. Is it supported?

**Comment by Baron-von-Riedesel** [comment]
> Just wanted to add that the sound card is SigmaTel. Is it supported?

HDA is - unlike AC97 - a "full" standard, meaning that no special care should be needed for any specific implementation.

That doesn't mean that HDAPLAY is supposed to work identically on any HDA device. The DACs may support certain frequencies only ( 11025 or 22050, for example, are often NOT supported ); or the HDA device is digital only ( without any DAC ). In such cases, HDAPLAY has to give up. However, for your machine this isn't true.

**Comment by tiagotarifa** [comment]
> HDAPLAY wants to access the HDA controller exclusively - hence cannot run concurrently with SBEMU/VSBHDA.
> 
> I checked your sample1.zip - it's not really a "beep", sounds more like a simple sinusoidal wave.
> 
> btw, lots of tools ( i.e. audacity ) can be used to convert any MP3 to WAV, stereo, 16bit, 44.1 kHz.
> 
> There are no more tests that I can supply.

I made this test with a [WAV stereo 16bit 44.1 kHz file](https://dl.espressif.com/dl/audio/gs-16b-2c-44100hz.wav).
It partially works using `hdaplay.exe -s file.wav`. But it's plays a poor sound. It was look like a 8bit nes sound playing (like castlevania song).

![tela3](https://github.com/Baron-von-Riedesel/VSBHDA/assets/17298749/ffad30c2-d2c4-45c7-9604-a7732e6d7817)

I made tests using vsbhda too. I tried the -s (0, 1 and 2) option ether but no sound. Doom, biomenace and Epic Pinball was used.

Another sample wav files could be found here:
https://espressif-docs.readthedocs-hosted.com/projects/esp-adf/en/latest/design-guide/audio-samples.html
https://www.mmsp.ece.mcgill.ca/Documents/AudioFormats/WAVE/Samples.html
